### PR TITLE
Fix for minimum-peer-connectivity and arg consistency changes

### PR DESCRIPTION
**Pull request by will-gluwa**
- Respect minimum-peer-connectivity value: Fixes [STL-1709](https://jira.hyperledger.org/browse/STL-1709)
- Rename args for consistency

### CODE REVIEW

File: validator/src/server/cli.rs [hunk 1]
```diff
@@ -44,11 +44,11 @@ pub fn wrap_in_pydict(py: Python, matches: &ArgMatches) -> PyResult<PyDict> {
         py,
         "minimum_peer_connectivity",
         matches
-            .value_of("minimum-peer-connectivity")
+            .value_of("minimum_peer_connectivity")
```
Comment (arsulegai) [inline validator/src/server/cli.rs:47]: This is the CLI parameter for rust version, it then is mapping to python dictionary `minimum_peer_connectivity`. Can you confirm which version of Sawtooth you're currently on?

File: validator/src/server/cli.rs [hunk 2]
```diff
@@ -221,15 +221,15 @@ pub fn parse_args<'a>() -> ArgMatches<'a> {
     app.get_matches()
 }
 
-#[allow(clippy::needless_pass_by_value)]
+#[allow(needless_pass_by_value)]
```
Comment (arsulegai) [inline validator/src/server/cli.rs]: This change in the commit is unnecessary. Solution: Squash the other commit that is reverting this part into the same commit.

**Review by arsulegai — CHANGES_REQUESTED** [review]

File: validator/src/server/cli.rs (hunk 1 above)
Comment (will-gluwa) [reply validator/src/server/cli.rs:47]: Response from my colleague who implemented the fix in our form:

> Right, the issue is that the value for the parameter is stored under the key "minimum_peer_connectivity" (https://github.com/hyperledger/sawtooth-core/blob/7651cb5552800dbac98559b46339a84eae472ef7/validator/src/server/cli.rs#L187) but then the value is looked up by the key "minimum-peer-connectivity" (https://github.com/hyperledger/sawtooth-core/blob/7651cb5552800dbac98559b46339a84eae472ef7/validator/src/server/cli.rs#L47) which doesn't exist, so it always falls back to the default value of 3 (https://github.com/hyperledger/sawtooth-core/blob/7651cb5552800dbac98559b46339a84eae472ef7/validator/sawtooth_validator/config/validator.py#L201).

Comment (arsulegai) [reply validator/src/server/cli.rs:47]: Thanks for the explanation. I see inconsistency there. We may need to back-port and release a patch if required.

**Review by arsulegai — APPROVED** [review]

**Comment by arsulegai** [comment]
@rberg2 @rbuysse user @will-gluwa needs to be added in authorized list for CI.

**Comment by will-gluwa** [comment]
Any update?

**Comment by agunde406** [comment]
We merged in a fix for the documentation build last week. Can you rebase this PR to pick up that build fix?

**Comment by will-gluwa** [comment]
@agunde406 I merged `hyperledger:main` into my branch. Hopefully that's what you required.

**Comment by arsulegai** [comment]
> @agunde406 I merged `hyperledger:main` into my branch. Hopefully that's what you required.

@will-gluwa please rebase instead. Keep the original commit alone in the PR.

**Comment by will-gluwa** [comment]
> > @agunde406 I merged `hyperledger:main` into my branch. Hopefully that's what you required.
> 
> @will-gluwa please rebase instead. Keep the original commit alone in the PR.

@agunde406 Did rebase based on https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request, please let me know if I did it correctly

**Review by arsulegai — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Comment by agunde406** [comment]
I added two backport PRs for 1-2 and 1-3

**Pull request merged**